client-core/src/synchronizer.rs

```diff
@@ -19,6 +20,9 @@ type WalletPassphrase = SecKey;
 pub struct PollingSynchronizer {
     wallets: Arc<Mutex<BTreeMap<WalletName, WalletPassphrase>>>,
     progress: Arc<Mutex<SynchronizerProgress>>,
+    should_run: Arc<AtomicBool>,
+    sync_thread: Option<thread::JoinHandle<()>>,
```

Comment:
how about using one mutx for wallets, progress, should_run

Reply:
No, they are for different purposes.